modules/gapi/include/opencv2/gapi/cpu/video.hpp

```diff
@@ -17,9 +17,39 @@ namespace cpu {
 GAPI_EXPORTS GKernelPackage kernels();
 
 } // namespace cpu
+
+struct BackgroundSubtractorParams
+{
+    int history;
+    double threshold;
+    bool detectShadows;
+
+    BackgroundSubtractorParams()
+    {
+        history = 500;
+        threshold = 16;
+        detectShadows = true;
+    }
+
+    BackgroundSubtractorParams(int hist, double thr, bool detect)
+    {
+        history = hist;
+        threshold = thr;
+        detectShadows = detect;
+    }
+};
```

Comment:
Please, place the implementation to .cpp file
I suugest `modules/gapi/src/backends/cpu/gcpuvideo.cpp` next to the kernel which uses this

Reply:
Reworked. So it outdated.